src/ngraph/op/util/arithmetic_reduction.hpp

```diff
@@ -37,32 +37,19 @@ namespace ngraph
                 /// \param arg Output that produces the first input tensor.
                 /// \param reduction_axes The axis positions (0-based) to be eliminated.
                 ArithmeticReduction(const Output<Node>& arg, const AxisSet& reduction_axes);
-
-                /// \brief Constructs an arithmetic reduction operation.
-                ///
-                /// \param arg Node that produces the first input tensor.
-                /// \param reduction_axes The axis positions (0-based) to be eliminated.
-                ArithmeticReduction(const std::shared_ptr<Node>& arg,
-                                    const AxisSet& reduction_axes);
-
                 /// \brief Constructs an arithmetic reduction operation.
                 ///
-                /// \param arg Node that produces the first input tensor.
+                /// \param arg Output that produces the first input tensor.
                 /// \param reduction_axes The axis positions (0-based) to be eliminated.
-                ArithmeticReduction(const std::string& node_type,
-                                    const std::shared_ptr<Node>& arg,
-                                    const AxisSet& reduction_axes);
+                ArithmeticReduction(const Output<Node>& arg, const Output<Node>& reduction_axes);
 
             public:
                 void validate_and_infer_types() override;
 
                 /// \return The axis positions (0-based) to be eliminated through reduction.
-                const AxisSet& get_reduction_axes() const { return m_reduction_axes; }
+                const AxisSet get_reduction_axes() const;
```

Comment:
Typically, compilers might optimize the copy but we have been using the return by ref convention for getters on other ops.

Reply:
Sure. Since we now return by value in this case, the const can prevent some cases where you might want to use the return of the getter as an r-value temporary. Certainly don't remove const on any getters that are returning an l-value reference, but I guess I don't see how the const on the return-by-value is helpful.